stock_intercompany/models/stock_move.py

```diff
@@ -0,0 +1,17 @@
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _action_assign(self):
+        """
+        Override to create counterpart pickings after moves are assigned.
+        """
+        res = super()._action_assign()
+
+        counterparts = self.mapped("picking_id")._create_counterpart_pickings("out")
```

Comment:
mapped is not required you can directly use self.picking_id (and it will map all picking_id or moves in self)

```suggestion
        counterparts = self.picking_id._create_counterpart_pickings("out")
```